### PR TITLE
Add store's action queue state on ViewStore and modify private only setter

### PR DESCRIPTION
**Pull request by sobabear**
### Related Issues 💭
I have two suggestion's for ViewStore.

One is making a limitation for allowing setter on public.
The other one is let viewstore be able to know of store's state, especially actionQueue.

### Checklist ✅

- [x] If the changes affect existing functionality, please verify whether the above information has been appropriately described.

### CODE REVIEW

File: Sources/OneWay/ViewStore.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ where R.Action: Sendable, R.State: Sendable & Equatable {
     public let initialState: State
 
     /// The current state of a store.
-    public var state: State {
+    public private(set) var state: State {
```
Comment (DevYeom) [inline Sources/OneWay/ViewStore.swift:29]: It is so reasonable.
Could you also set the `state` of `Store` to `public private(set)`?

**Review by DevYeom — COMMENTED** [review]
What you first suggested was necessary. Thanks!
However, I don't understand the need for your second suggestion about `actionQueue` and `isProcessing`. Could you explain why it is needed?

> [!NOTE]
> Directly modifying the state of the store can lead to unintended side effects and is considered an anti-pattern. It is preferable to change the state only by calling `send()`.

**Comment by sobabear** [comment]
@DevYeom 
Thanks to reply!

The reason of second one, just my opinion, is that i thought Combination with 'queue-based' system and Unidirectional system could be let know developers proper informations of `state`. 

I already know  `Store` does automatically manage actions(queue based) so user(developer) does not need to handle it at all. Moreover some syntax shall help it such as `states: AsyncStream`. On the others hands when handle very long timed action (which let store is on processing and actionqueue is not empty), it would be very helpful to developer to know least information of store's state

**Comment by DevYeom** [comment]
Your second suggestion also makes sense. However, There is a way to find out if a long-duration action is being processed could be as follows.

```swift
func reduce(state: inout State, action: Action) -> AnyEffect<Action> {
// ...
    case .request:
        return .concat(
            .just(.setIsLoading(true)),
            .single {
                let result = await longTimeAPI.request()
                return .response(result)
            },
            .just(.setIsLoading(false))
        )
    case .setIsLoading(let isLoading):
        state.isLoading = isLoading
        return .none
    }
// ...
}
```

I also have a plan to support a debug print feature(e.g. printing changes in state to the console). I hope you understand that I try not to expose the interface externally(`public`) as much as possible to reduce misuse.

How about only including the first suggestion this time?

**Comment by sobabear** [comment]
> Your second suggestion also makes sense. However, There is a way to find out if a long-duration action is being processed could be as follows.
> 
> ```swift
> func reduce(state: inout State, action: Action) -> AnyEffect<Action> {
> // ...
>     case .request:
>         return .concat(
>             .just(.setIsLoading(true)),
>             .single {
>                 let result = await longTimeAPI.request()
>                 return .response(reulst)
>             },
>             .just(.setIsLoading(false))
>         )
>     case .setIsLoading(let isLoading):
>         state.isLoading = isLoading
>         return .none
>     }
> // ...
> }
> ```
> 
> I also have a plan to support a debug print feature(e.g. printing changes in state to the console). I hope you understand that I try not to expose the interface externally(`public`) as much as possible to reduce misuse.
> 
> How about only including the first suggestion this time?

Very acceptable decision! Please check it out! @DevYeom

**Review by DevYeom — APPROVED** [review]
Thanks!

**Pull request merged**